pkg/kinflate/util/util.go

```diff
@@ -50,6 +48,17 @@ func Decode(in []byte, into types.KObject) (types.KObject, error) {
 	if err != io.EOF {
 		return nil, err
 	}
+	return objs, nil
+}
+
+// DecodeToKObject decodes a list of objects in byte array format.
+// Decoded object will be inserted in `into` if it's not nil. Otherwise, it will
+// construct a new map and return it.
+func DecodeToKObject(in []byte, into types.KObject) (types.KObject, error) {
```

Comment:
Just curiosity, do we plan on keeping `KObject`?

Reply:
I don't think we need to. The type resource defined in #313 has a method GVKN(). With that, we can get the same info stored in KObject.